### PR TITLE
Updates before Cohort 11 start

### PR DESCRIPTION
**Pull request by shrutivanw**
Summary of changes made:
- Mention and link to appropriate JumpStart Live lessons
- Consistently use "JumpStart" as the name
- Consistenty call ourselves "Ada Developers Academy" instead of just "Ada"
- Don't have examples using loops until loops are introduced in the curriculum
- Don't show diagram with array indices until after indices are introduced.
- Avoid mentioning or drawing comparisons with concepts not introduced in this content: methods, Python comparisons etc.
- Do not use variable names that are method names e.g. `p` in our examples.
- Mention about classes, objects and methods as an aside. (Address consistent JSL student feedback on learning sooner that almost everything in Ruby is a class with methods available).
- Add a numbers' problem assignment that evolves as we introduces new concepts through out JumpStart.
- Add a planets in the solar system assignment for lessons on arrays and hashes.
- Fit and finish clean up, remove unneeded files.

### CODE REVIEW

File: learning-to-code/hashes/README.md [hunk 1]
```diff
@@ -10,119 +10,274 @@ _Jump start: Lesson 13_
...
-my_dog["name"] # => "barkly"
+# your_dog is defined using the symbol notation above
+puts your_dog[:age] # 2
+puts your_dog[:breed] # beagle
+puts your_dog[:name] # barkly
 
 # by default, will return nil for keys that don't exist
-my_dog["color"] # => nil
+puts my_dog[:color] # nil
 ```
```
Comment (droberts-ada) [inline learning-to-code/hashes/README.md]: I really like that we're now using symbols as the default.

File: learning-to-code/arrays/README.md [hunk 2]
```diff
@@ -48,40 +51,63 @@ We can utilize the syntax for both of the above to initialize a new Array _with
...
-Arrays can store all sorts of data. Usually, it makes the most sense to have a single data type in an array, but in Ruby, it is not a requirement.
 
-This means that arrays can store data types of `Integer`, `String`, and `Float`!
+```ruby
+numbers = [1, 2, 3, 4]
+puts numbers[2] # => 3
+```
+| Value |  1  |  2 |  3 |  4 |
+| :---- | :-- | :-- | :-- | :-- |
+| Index | [0] | [1] | [2] | [3] |
```
Comment (droberts-ada) [inline learning-to-code/arrays/README.md]: Good call making these tables instead of images. Much more maintainable.

File: learning-to-code/arrays/README.md [hunk 3]
```diff
@@ -162,3 +195,95 @@ cats = ["grumpy"]
...
+
+   The minimum value in the array is 21.
+   The maximum value in the array is 582.
+   The average of all the values in the array is 159.4.
+```
+
+## Planet Information Assignment
+* Create two _synchronized_ arrays.
+  * In one array, save the names of the planets in our solar system in the order of their distance from the sun: Mercury, Venus, Earth, Mars, Jupiter, Saturn, Uranus, Neptune.
+  * In the other array, save the radius of each planet at the corresponding index.
```
[the remark this answers is not in the document]
Comment (shrutivanw) [reply learning-to-code/arrays/README.md]: Thank you for calling that out! It wasn't intentional. Let me change that assignment.

Humm.. I hope we don't cover continents anywhere. :-)

File: learning-to-code/arrays/README.md [hunk 4]
```diff
@@ -48,40 +51,63 @@ We can utilize the syntax for both of the above to initialize a new Array _with
...
+default_array = Array.new(3, "def")
+puts default_array # => ["def", "def", "def"]
+```
+| Value | "def" |  "def" |  "def" |
+| :---- | :-- | :-- | :-- |
+| Index | [0] | [1] | [2] |
+
+
+### Data Types
+Arrays can store all sorts of data. In most programming languages, arrays are _homogeneous_ data structures. This means that in most programming lanaguages, the same type of data is stored in each element of the array. e.g. an array of integers, an array of strings etc. However, in Ruby, it is not a requirement for all elements in the array to be of the same type.
```
[the remark this answers is not in the document]
Comment (shrutivanw) [reply learning-to-code/arrays/README.md]: I hear you. We may not need to get into the details during JumpStart. I will reword it to focus only on how Ruby does it in the JumpStart curriculum.

*Thoughts for offline discussion*: Keeping in mind the idea of learning how to learn a new language (since a lot of the students will have to learn Java, C++ or C# during internship), and to make it easier for the students to grasp data structure concepts later, my motive here was to call out that Ruby's Array class is a special, language specific construct. 

1. The Array class in Ruby does not follow the same rules as an Array data structure. [Array data structure](https://www.geeksforgeeks.org/overview-of-data-structures-set-1-linear-data-structures/): data structure used to store homogeneous elements at contiguous locations. Arrays are also static data structures, unlike Ruby's Array class which can dynamically grow and shrink in size.
2. There are other C like languages (C, C++, C#, Java), where arrays of primitive data types always mean homogeneous elements at contiguous location. Some of the higher level languages like C# and Java provide other classes like ArrayList, which provide more flexibility like Ruby's Array class provides.
This gets covered in the second CS Fundamental's class. Are there other places in the curriculum where we could underscore this again?

File: learning-to-code/arrays/README.md [hunk 5]
```diff
@@ -97,58 +123,65 @@ random = [7, "fourteen", 13.2, "horse", 1]
...
+puts animals[5]      # => "lizard"
 ```
 
-**first**
+**first**  
 This method will return the first element, or the element at the zero index, in an array.
 
-**last**
-This method will return the last element in the array. The index of this element will depend solely on how many items are in the array.
+**last**
```
[the remark this answers is not in the document]
Comment (shrutivanw) [reply learning-to-code/arrays/README.md]: Good call out! Thanks - now done.

File: learning-to-code/hashes/README.md [hunk 6]
```diff
@@ -10,119 +10,274 @@ _Jump start: Lesson 13_
...
-By using the `new` syntax and specifying a default value, _all_ keys retrieved with no corresponding value will return the default value. Without the default value, accessing an undefined key will return `nil`.
+A new hash can be created with key-value pairs populated as follows:
+```ruby
+my_dog = {
+  "name" => "woofy",
+  "breed" => "spaniel",
+  "age" => 3
+}
+```
+The above hash is called `my_dog`. It has three key-value pairs. All the keys are string objects. The values associated with `"name"` and `"breed"` are string objects, `"barkly"` and `"beagle"` respectively. The third key, `"age"` has a value of `2`, which is of type integer. In general, the value may be a more complex data structure and of any data type.
```
[the remark this answers is not in the document]
Comment (shrutivanw) [reply learning-to-code/hashes/README.md]: Good catch!

File: learning-to-code/hashes/README.md [hunk 7]
```diff
@@ -10,119 +10,274 @@ _Jump start: Lesson 13_
...
 ```ruby
-my_dog.values #=> ["barkly", "beagle", 3]
+puts your_dog.values #=> ["barkly", "beagle", 2]
 ```
 
+
+## Planet Information Assignment
+* Create a hash to maintain the information of the right planets in our solar system.
+  * For each planet, save the following information:
+    * Name of the planet
```
[the remark this answers is not in the document]
Comment (shrutivanw) [reply learning-to-code/hashes/README.md]: It was intentional. IMHO, the numbers' problem assignment in hashes was harder than planet's assignment in hashes. It was the reverse case with arrays.

I'm removing the planets in the solar system assignments from this PR. I'll send a new PR later with a new continents assignment.

File: learning-to-code/hashes/README.md [hunk 8]
```diff
@@ -10,119 +10,274 @@ _Jump start: Lesson 13_
...
+Thank you for joining us on learning about the planets.
+```
+
+
+## Numbers problem assignment
+* Throughout this course we will be working with the **Numbers program**. Here's the fifth version of the program:
+  * Leveraging your learnings from the notes you read (different ways of organizing data in arrays and hashes), write a program that does the following:
+    * Create a hash.
+      * Add one key-value pair in the hash. The value should be an array of five randomly generated integers between 12 and 21, both inclusive.
+      * Create another key-value pair in the hash. The value should be an array of five positive integers that the program receives as user input, also in the range of 12 and 21, both inclusive.
```
[the remark this answers is not in the document]
Comment (shrutivanw) [reply learning-to-code/hashes/README.md]: Good catch for missing information about keys. Now fixed.

File: learning-to-code/hashes/assignments/hash-worksheet.md [hunk 9]
```diff
@@ -39,14 +40,14 @@ puts animals["donkey"]
 
 3.
 ```ruby
-chuck_norris = {
+matrix_person = {
   "punch" => 99,
   "kick" => 98,
   "stops_bullets_with_hands" => false
```
[the remark this answers is not in the document]
Comment (shrutivanw) [reply learning-to-code/hashes/assignments/hash-worksheet.md]: Good call! Done.

File: learning-to-code/hashes/assignments/hash-worksheet.md [hunk 10]
```diff
@@ -60,5 +61,79 @@ puts menu["burger"]
...
+
+puts person.length
+puts person[:last_name]
+```
+
+6.
+```ruby
+animal_kingdom = {
+  mammals: ["rats", "dogs", "cats", "humans"],
+  reptiles: ["alligator", "crocodile"],
```
[the remark this answers is not in the document]
Comment (shrutivanw) [reply learning-to-code/hashes/assignments/hash-worksheet.md]: The JumpStart curriculum doesn't yet talk about nested data structures. I agree with you, we should introduce the concept in the curriculum before adding it to the worksheet. I would like to introduce it in JumpStart. I'll remove it for now and leave it for future PR.

Thanks for the link to the textbook resource. Note to self: I should go through week 1 videos.

File: learning-to-code/iterators/README.md [hunk 11]
```diff
@@ -272,3 +212,130 @@ end
...
+     multiple = i * 10
+     # multiple is scoped to this block
+     puts "#{i} times 10 is #{multiple}"
+     i += 1
+   end
+   # multiple is an undefined variable
+```
+
+In Ruby, one way to identify some of the blocks is to identify code statements surrounded by either `do ... end` or `{ }`.
+
```
[the remark this answers is not in the document]
Comment (shrutivanw) [reply learning-to-code/iterators/README.md]: Done.

File: learning-to-code/iterators/README.md [hunk 12]
```diff
@@ -272,3 +212,130 @@ end
 
 puts "You guessed it!"
 ```
+
+### Each
+The `each` iterator is used on a collection of data, like a Range, Array or Hash. We have already come across Range and we will learn more about Arrays and Hashes in later lessons. This iterator will iterate over each element of the collection, one by one, and provide access to each element using the _iteration variable_. The `each` iterator will continue to execute until it has reached the end of the collection.
+
+The `each` iterator is comprised of three pieces:
```
[the remark this answers is not in the document]
Comment (shrutivanw) [reply learning-to-code/iterators/README.md]: That makes sense. I left the portion describing `.each` iterators on range values in the notes for the iterators' lesson, and moved the remaining examples and explanations related to arrays to the lesson notes on arrays.

File: learning-to-code/programming-expressions/README.md [hunk 13]
```diff
@@ -128,21 +137,40 @@ end
...
+```
+
 ## Candy machine assignment
 
 ### Overview
 This challenge is to build a computer candy machine. You’ve got some money and you want to buy some candy.  So, you go up to the candy machine, put in your money, select your candy, and then pick up your candy and your change. You can use `gets.chomp` to get input from the user without the extra newline at the end.
 
 ### Requirements
-* Ask the user how much money they have, assume that the `$` symbol is part of the prompt (the user doesn't have to enter it)
+* Ask the user how much money they have, assume that thUser input is indicated in ~~ (tildes).e `$` symbol is part of the prompt (the user doesn't have to enter it)
```
[the remark this answers is not in the document]
Comment (shrutivanw) [reply learning-to-code/programming-expressions/README.md]: Complete oversight on my part! Now fixed.

**Comment by shrutivanw** [comment]
Thank you, Dan. All great feedback. I'll be mindful of smaller commits and smaller PRs in the future.
Based on the code review discussion and new updates, I'm making notes to do the following in future PRs:
Future work on JumpStart:

1. Add a new assignment in Arrays & Hashes (Continents assignment instead of proposed planets in the solar system assignment).
2. Extend notes in hashes to include example use of .each iterator.
3. Extend JumpStart to promote thinking of nested data structures.

File: learning-to-code/arrays/README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (droberts-ada) [reply learning-to-code/arrays/README.md]: It's definitely an idea that we want them to get to eventually - I think early in the CS fun curriculum is a good spot for this.
